Sources/OpenCombine/Publishers/Publishers.Decode.swift

```diff
@@ -0,0 +1,108 @@
+
+        public let upstream: Upstream
+
+        internal let decoder: Coder
+
+        public init(upstream: Upstream, decoder: Coder) {
+            self.upstream = upstream
+            self.decoder = decoder
+        }
+
+        /// This function is called to attach the specified `Subscriber`
+        /// to this `Publisher` by `subscribe(_:)`
+        ///
+        /// - SeeAlso: `subscribe(_:)`
+        /// - Parameters:
+        ///     - subscriber: The subscriber to attach to this `Publisher`.
+        ///                   once attached it can begin to receive values.
+        public func receive<Receiver: Subscriber>(subscriber: Receiver)
+            where Failure == Receiver.Failure, Output == Receiver.Input {
+                let decodeSubscriber = _Decode<Upstream, Receiver, Coder>(
+                    downstream: subscriber,
+                    decoder: decoder
+                )
+                upstream.receive(subscriber: decodeSubscriber)
+        }
+    }
+}
+
+// swiftlint:disable:next line_length
+private final class _Decode<Upstream: Publisher, Downstream: Subscriber, Coder: TopLevelDecoder>:
+    OperatorSubscription<Downstream>, Subscriber,
+    CustomStringConvertible, Subscription
+    where
+    Downstream.Input: Decodable,
+    Coder.Input == Upstream.Output,
+    Downstream.Failure == Error {
+
+    typealias Input = Upstream.Output
+    typealias Failure = Upstream.Failure
+    typealias Output = Downstream.Input
+
+    private let _decoder: Coder
+    private var _demand: Subscribers.Demand = .none
+
+    var description: String { return "Decode" }
+
+    init(downstream: Downstream, decoder: Coder) {
+        self._decoder = decoder
+        super.init(downstream: downstream)
+    }
+
+    func receive(subscription: Subscription) {
+        upstreamSubscription = subscription
+        subscription.request(.unlimited)
+        downstream.receive(subscription: self)
+    }
+
+    func receive(_ input: Input) -> Subscribers.Demand {
+        do {
+            let value = try _decoder.decode(Downstream.Input.self, from: input)
+            return downstream.receive(value)
+        } catch {
+            downstream.receive(completion: .failure(error))
+            cancel()
+            return .none
+        }
+    }
+
+    func receive(completion: Subscribers.Completion<Failure>) {
+        switch completion {
+        case .finished:
+            downstream.receive(completion: .finished)
+        case .failure(let error):
+            downstream.receive(completion: .failure(error))
+        }
+    }
+
+    func request(_ demand: Subscribers.Demand) {
+        _demand = demand
+    }
+}
+
+extension Publisher {
+    public func decode<Item, Coder>(type: Item.Type, decoder: Coder)
+        -> Publishers.Decode<Self, Item, Coder>
+        where Item : Decodable, Coder : TopLevelDecoder, Self.Output == Coder.Input {
+            return Publishers.Decode(upstream: self, decoder: decoder)
```

Comment:
Let's fix indentation here as well, for example see
https://github.com/broadwaylamb/OpenCombine/blob/8b7bb88d66c4570cb37070d19d8f1f1f8735f8f5/Sources/OpenCombine/Publishers/Publishers.Multicast.swift#L53-L59

Reply:
This line is indented with 8 spaces instead of 4